### PR TITLE
Add some security

### PR DESCRIPTION
**Pull request by lyio**
This is obviously just a proof of concept...If the concept is accepted I'll clean this up and make it pretty.

Also, the frontend needs to write the list first, obviously.
- Use firebase instead of post body to retrieve the grocery items that need to be sent.
- Evaluate Auth header to prevent anauthorized access

### CODE REVIEW

File: app/logic/util.js [hunk 1]
```diff
@@ -1,5 +1,27 @@
+firebase = require('firebase');
+
 module.exports = {
   /**
+   * Extract the items from the firebase backend using the userId
+   * 
+   * @param {String} userId
+   */
+  getGroceriesForUser: function(userId) {
+      var ref = new Firebase("https://socobo-dev-project.firebaseio.com/grocery-list/" + userId);
```
Comment (flashback2k14) [inline app/logic/util.js]: this route doesn't get the active items list.

**Comment by flashback2k14** [comment]
check for an socobo auth header is a good idea and should be implemented, but why we should call firebase for the items? we have all needed infos in the frontend.

**Comment by patrick-hebner** [comment]
I think he wants to request the items because this would ensure the specified AUTH token belongs to a real user in the db, right? If you would only check the existence of the token an then send the content provided by the request body, it would be enough to provide any token.

To sum up, it would be necessary to verify the provided token against the users and the rules of our firebase backend.

If I'm wrong correct me please! That is just my understanding of the PR after a quick first scan.

File: app/logic/util.js (hunk 1 above)
Comment (lyio) [inline app/logic/util.js]: That is correct. It made sense to me to create a separate entry in the database so the access can be more strictly limited.

**Comment by lyio** [comment]
@hetec that was the reasoning behind that. It would be even better to not use the supplied mail address but get that from the database as well.

With only requiring an auth header you could still send anything. I tried to make sure here that you'd at least have to correctly guess a valid user id.

More security would of course require some more changes. For example, I enabled anonymous access to the firebase. The backend would have to get its own login and then there could be a node in Firebase where only this specific user has read access.

**Comment by flashback2k14** [comment]
if you only send the userid, than you need a call on the profile route to get the user email and than a call to get the grocery list. for me, this is too much for simply send an list of items to the user.

File: app/logic/util.js (hunk 1 above)
Comment (flashback2k14) [inline app/logic/util.js]: ?

**Comment by lyio** [comment]
Authenticating the server could be done via the [app secret](https://www.firebase.com/docs/rest/guide/user-auth.html)

And yes, that would be two calls.

Since I mainly want to prevent spamming of random email addresses via our backend, retrieving the address from the database and taking the items from the post request would accomplish the same thing and only be one request.

**Comment by flashback2k14** [comment]
OK, than implement this behavior and we can See if it work.

**Comment by flashback2k14** [comment]
FYI debugging with vs code works like a charm. :-)

**Comment by lyio** [comment]
The backend now requires the Firebase app secret as `FIREBASE_TOKEN` in the environment.

**Comment by flashback2k14** [comment]
I have tried The code, but it didn't worked for me. With curl i get The E-Mail address, but not with The https request.

**Comment by flashback2k14** [comment]
lgtm

**Comment by flashback2k14** [comment]
sorry, i have mistakenly merged the PR.

**Pull request merged**